File: src/main/java/com/cag/cagbackendapi/entities/TrainingEntity.java
package com.cag.cagbackendapi.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import java.util.UUID;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "training")
public class TrainingEntity {
    @Id
    @Type(type = "pg-uuid")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator"
    )
    private UUID training_id;
    private String institution;
    private String degree;
    private short start_year;
    private short end_year;
    private String country;
    private String city;
    private String state;
    private String notes;

    @OneToOne
    @JoinColumn(foreignKey = @ForeignKey(name = "fk_profile"))
    private ProfileEntity profileEntity;
}

File: src/main/java/com/cag/cagbackendapi/entities/TrainingEntity.kt
package com.cag.cagbackendapi.entities

import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.Type
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "training")
data class TrainingEntity (
    @Id
    @Type(type = "pg-uuid")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator"
    )
    var training_id: UUID?,
    var institution: String?,
    var degree: String?,
    var start_year: Short?,
    var end_year: Short?,
    var country: String?,
    var city: String?,
    var state: String?,
    var notes: String?,

    @OneToOne
    @JoinColumn(name = "profile_id", referencedColumnName = "profile_id")
    var profileEntity: ProfileEntity?
)
